src/component/card/VideoCard.jsx

```diff
@@ -28,19 +30,30 @@ export const VideoCard = (video) => {
     views,
     playlistId,
   } = video;
+  const { isLoggedIn } = useSelector((store) => store.authTimeline);
   const { likes } = useSelector((store) => store.likeTimeline);
   const { watchLater } = useSelector((store) => store.watchLaterTimeline);
+  const { history } = useSelector((store) => store.historyTimeline);
   const { pathname } = useLocation();
   const dispatch = useDispatch();
 
   const isLiked = likes.some((like) => like._id === _id);
-  const isInWatchLater = watchLater.some(
+  const isVideoInWatchLater = watchLater.some(
     (watchLaterVideo) => watchLaterVideo._id === _id
   );
+  const isVideoInHistory = history.some(
+    (historyVideo) => historyVideo._id === _id
+  );
 
   return (
     <article className="md:w-[18rem] lg:w-[20.5rem] hover:shadow-md hover:shadow-violet-700/50">
-      <Link to={`/video/${_id}`} className="relative">
+      <Link
+        to={`/video/${_id}`}
+        className="relative"
+        onClick={() =>
+          isLoggedIn && !isVideoInHistory && dispatch(postToHistory(video))
+        }
```

Comment:
Good logic of using History dispatch action only at the time when the user is getting logged in!!